### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by alexmt**
PR has following CI improvements:

- enables auth in circle ci: https://github.com/argoproj/argo-cd/pull/1667/files#diff-1d37e48f9ceff6d8030570cd36286a61L278
- remove hardcoded 60 seconds timeout: https://github.com/argoproj/argo-cd/pull/1667/files#diff-1d37e48f9ceff6d8030570cd36286a61R286
- It appears that each `kubectl` command takes at least 100ms. Tried to switch to k8s api and it saved 40 seconds: https://github.com/argoproj/argo-cd/pull/1667/files#diff-908ea2b2aac0dffff86c703e913175b4L140 
- fixes test fixture initialization bug: https://github.com/argoproj/argo-cd/pull/1667/files#diff-908ea2b2aac0dffff86c703e913175b4R153

### CODE REVIEW

**Review by jessesuen — APPROVED** [review]

File: test/e2e/fixture/app/consequences.go [hunk 1]
```diff
@@ -52,15 +51,7 @@ func (c *Consequences) app() *Application {
...
-	output, err := fixture.RunCli("app", "get", c.context.name, "-o", "yaml")
-	if err != nil {
-		if strings.Contains(output, "NotFound") {
-			return nil, nil
-		}
-		return nil, err
-	}
-	app := &Application{}
-	return app, yaml.Unmarshal([]byte(output), app)
+	return fixture.AppClientset.ArgoprojV1alpha1().Applications(fixture.ArgoCDNamespace).Get(c.context.name, v1.GetOptions{})
```
Comment (alexec) [inline test/e2e/fixture/app/consequences.go]: I know this looks odd, but the intent to was make sure that we prefer to invoke the CLI tool, than the API. How come you changed this bit?

File: test/e2e/fixture/fixture.go [hunk 2]
```diff
@@ -63,10 +64,6 @@ func getKubeConfig(configPath string, overrides clientcmd.ConfigOverrides) *rest
 // creates e2e tests fixture: ensures that Application CRD is installed, creates temporal namespace, starts repo and api server,
 // configure currently available cluster.
 func init() {
-
-	// trouble-shooting check to see if this busted add-on is going to cause problems
-	FailOnErr(Run("", "kubectl", "api-resources", "-o", "name"))
```
Comment (alexec) [inline test/e2e/fixture/fixture.go]: 👍

File: test/e2e/fixture/fixture.go [hunk 3]
```diff
@@ -136,22 +133,15 @@ func EnsureCleanState() {
...
-	}
-	FailOnErr(Run("", "kubectl", "delete", "appprojects", "--field-selector", "metadata.name!=default"))
-	// takes around 5s, so we don't wait
+	CheckError(AppClientset.ArgoprojV1alpha1().Applications(ArgoCDNamespace).DeleteCollection(&v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{}))
+	CheckError(AppClientset.ArgoprojV1alpha1().AppProjects(ArgoCDNamespace).DeleteCollection(
+		&v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{FieldSelector: "metadata.name!=default"}))
+	CheckError(KubeClientset.CoreV1().Secrets(ArgoCDNamespace).DeleteCollection(
+		&v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{LabelSelector: testingLabel + "=true"}))
+
 	FailOnErr(Run("", "kubectl", "delete", "ns", "-l", testingLabel+"=true", "--field-selector", "status.phase=Active", "--wait=false"))
```
Comment (alexec) [inline test/e2e/fixture/fixture.go]: Do you need to delete this line too?

File: test/e2e/fixture/fixture.go [hunk 4]
```diff
@@ -160,6 +150,7 @@ func EnsureCleanState() {
 		// changing theses causes a restart
 		AdminPasswordHash:    s.AdminPasswordHash,
 		AdminPasswordMtime:   s.AdminPasswordMtime,
+		ServerSignature:      s.ServerSignature,
```
Comment (alexec) [inline test/e2e/fixture/fixture.go]: 👍

File: test/e2e/fixture/fixture.go [hunk 5]
```diff
@@ -136,22 +133,15 @@ func EnsureCleanState() {
...
-			appName := strings.TrimPrefix(name, "application.argoproj.io/")
-			// we cannot delete any app, if an op is in progress
-			_, _ = RunCli("app", "terminate-op", appName)
-			// is it much more reliable to get argocd to delete an app than kubectl, deleting directly can make ArgoCD stuck
-			_, _ = RunCli("app", "delete", appName)
-		}
-	}
-	FailOnErr(Run("", "kubectl", "delete", "appprojects", "--field-selector", "metadata.name!=default"))
-	// takes around 5s, so we don't wait
+	CheckError(AppClientset.ArgoprojV1alpha1().Applications(ArgoCDNamespace).DeleteCollection(&v1.DeleteOptions{PropagationPolicy: &policy}, v1.ListOptions{}))
```
Comment (alexec) [inline test/e2e/fixture/fixture.go]: Maybe comment with the kubectl equivilant?

File: test/e2e/fixture/fixture.go [hunk 6]
```diff
@@ -136,22 +133,15 @@ func EnsureCleanState() {
...
 
+	policy := v1.DeletePropagationBackground
 	// delete resources
-	text, err := Run("", "kubectl", "get", "app", "-o", "name")
-	CheckError(err)
-	for _, name := range strings.Split(text, "\n") {
-		if name != "" {
-			appName := strings.TrimPrefix(name, "application.argoproj.io/")
-			// we cannot delete any app, if an op is in progress
-			_, _ = RunCli("app", "terminate-op", appName)
```
Comment (alexec) [inline test/e2e/fixture/fixture.go]: Do you need to maintain this line?

File: .circleci/config.yml [hunk 7]
```diff
@@ -275,17 +275,15 @@ jobs:
...
           environment:
             ARGOCD_FAKE_IN_CLUSTER: "true"
       - run:
           name: Wait for API server
           command: |
             set -x
-            # TODO - try to reduce to 20
-            sleep 60
-            curl -v --retry 5 localhost:8080
+            until curl -v http://localhost:8080/healthz; do sleep 3; done
```
Comment (alexec) [inline .circleci/config.yml]: nice

File: test/e2e/fixture/app/consequences.go (hunk 1 above)
Comment (alexmt) [reply test/e2e/fixture/app/consequences.go]: Forgot to mention it in PR description. This method is executed most frequently during e2e tests execution. Looks like any process fork takes at least 100ms, so I decided to switch it to api call to save several seconds

File: test/e2e/fixture/fixture.go (hunk 3 above)
Comment (alexmt) [reply test/e2e/fixture/fixture.go]: We still need to deleted temporal namespaces. There is no API to delete collection of namespaces, so I kept one `kubectl delete ...` . Would you prefer to use api here for consistency?

File: test/e2e/fixture/fixture.go (hunk 6 above)
Comment (alexmt) [reply test/e2e/fixture/fixture.go]: There is no need to terminate operation after fixing https://github.com/argoproj/argo-cd/issues/1665.  After application is deleted the controller is no longs stuck trying to update operation state.

Comment (alexec) [reply test/e2e/fixture/fixture.go]: 👍

File: test/e2e/fixture/fixture.go (hunk 3 above)
Comment (alexec) [reply test/e2e/fixture/fixture.go]: Nope, not fussed.

Comment (alexec) [reply test/e2e/fixture/fixture.go]: Hmmm.. How can there be no API for this, I'd assumed that kubectl just makes API calls?

File: test/e2e/fixture/fixture.go (hunk 5 above)
Comment (alexmt) [reply test/e2e/fixture/fixture.go]: sure. added

File: test/e2e/fixture/fixture.go (hunk 3 above)
Comment (alexmt) [reply test/e2e/fixture/fixture.go]: you are right. there is API, but no API to delete collection, only one by one deletion

**Pull request merged**